### PR TITLE
ci(cypress): adding the foundation for cypress integration tests & some starter coverage for login, search & updates

### PR DESCRIPTION
**Pull request by gabe-lyons**
Sets the groundwork for adding more cypress tests in the future. This PR gives an demonstration of how:

- cypress can be tied into CI to block prs when failing
- sample data can be loaded into the cypress environment
- tests that assert against sample data and mutate sample data can both be written
- examples of how utility methods (e.g. login and deleteUrn) can be written and referenced from tests

I also tested w/ a failing test earlier (before fix dataset name assertion) and verified that cypress outputs a very human-readable response when the test fails.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by gabe-lyons** [comment]
its running!

image

File: smoke-test/tests/cypress/cypress/fixtures/example.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "Using fixtures to represent data",
+  "email": "hello@cypress.io",
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/fixtures/example.json:3]: What is this?

Comment (jjoyce0510) [reply smoke-test/tests/cypress/cypress/fixtures/example.json:3]: (do we need it?)

File: smoke-test/tests/cypress/cypress/integration/mutations/mutations.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+describe('mutations', () => {
+  it('can create and add a tag to dataset and visit new tag page', () => {
+    cy.deleteUrn('urn:li:tag:CypressTestAddTag')
+    cy.login();
+    cy.visit('/dataset/urn:li:dataset:(urn:li:dataPlatform:hive,cypress_logging_events,PROD)');
+	cy.contains('cypress_logging_events');
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/integration/mutations/mutations.js:6]: nit: indent

Comment (jjoyce0510) [reply smoke-test/tests/cypress/cypress/integration/mutations/mutations.js:6]: or is this _indentional_?

File: smoke-test/tests/cypress/cypress/support/commands.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// existing commands.
+//
+// For more comprehensive examples of custom
+// commands please read more here:
+// https://on.cypress.io/custom-commands
+// ***********************************************
+//
+//
+// -- This is a parent command --
+Cypress.Commands.add('login', () => {
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/support/commands.js:13]: pretty cool!

File: smoke-test/tests/cypress/data.json [hunk 4]
```diff
@@ -0,0 +1,1560 @@
...
+                "time": 1581407189000,
+                "actor": "urn:li:corpuser:jdoe",
+                "impersonator": null
+              }
+            }
+          }
+        ]
+      }
+    },
+    "proposedDelta": null
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/data.json:1558]: more snapshots :(((((

File: smoke-test/tests/cypress/integration_test.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    delete_urns_from_file("tests/cypress/data.json")
+
+
+def test_run_cypress(frontend_session, wait_for_healthchecks):
+    command = f"npx cypress run"
+    print('starting?')
+    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, cwd="tests/cypress")
+    stdout = proc.stdout.read()
+    stderr = proc.stderr.read()
+    return_code = proc.wait()
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/integration_test.py:23]: fancyyy!

**Review by jjoyce0510 — APPROVED** [review]

File: smoke-test/tests/cypress/cypress/fixtures/example.json (hunk 1 above)
Comment (gabe-lyons) [reply smoke-test/tests/cypress/cypress/fixtures/example.json:3]: this is an example fixture generated by default by cypress- its just showing someone how they might add another fixture

**Review by shirshanka — APPROVED** [review]
LGTM

**Pull request merged**